docs/configuration/description.md

```diff
@@ -0,0 +1,69 @@
+Files are placed in the `.mainframer` directory.
+The final configuration is the combination of files placed in:
+
+* `${HOME}/.config/mainframer` or
+  [`${XDG_CONFIG_HOME}`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html)
+  if available — the global configuration;
+* `.mainframer` — the project configuration.
+
+It is posssible to declare common configuration using the global one
+and fine-tune it using the project one.
+
+## `.mainframer`
+
+The directory contains following files.
+
+Name         | Required | Keep in VCS | Description
+-------------|----------|-------------|------------
+`config.yml` | Yes      | No          | Configuration options.
+`ignore.yml` | No       | Yes         | Ignore rules for copying files.
+
+## `.mainframer/config.yml`
+
+```yaml
+remote:
+  name: "{SSH machine name}"
+push:
+  compression: {level}
+pull:
+  compression: {level}
+```
+
+Name               | Required | Value   | Default | Description
+-------------------|----------|---------|---------|------------------
+`remote.name`      | Yes      | `string`| —       | Remote machine name from SSH config or hostname / IP address.
+`push.compression` | No       | `0..9`  | `0`     | Compression level used to copy files from local machine to remote one.
+`pull.compression` | No       | `0..9`  | `0`     | Compression level used to copy files from remote machine to local one.
```

Comment:
default is 1

Reply:
Are these our values? Do you have a reference otherwise? I’ve found [an outdated email](http://samba.2283325.n4.nabble.com/what-is-the-default-value-for-compress-level-td2538905.html). [The source code says otherwise](https://github.com/kuna-systems/rsync/blob/5546dab32970955e77ef7a5886bcd8fb765a25bf/options.c#L1825-L1850) — seems to be `-1` (`Z_DEFAULT_COMPRESSION` value).

Reply:
MF `2.x` uses `1` by default for both cases.

Reply:
It's in 3.x PR #239, idea is that usually you push text files which are very compressible and pull binaries which are barely compressible, thus push `3` and pull `1`

Reply:
We can update this doc in #239, going to merge your PR now